File: src/main/java/io/pageobject/generator/locator/protractor/NgBindTemplateLocatorSource.java
package io.pageobject.generator.locator.protractor;

import io.pageobject.generator.GeneratorContext;
import io.pageobject.generator.attribute.HtmlAttributes;
import io.pageobject.generator.locator.AbstractLocatorSource;
import org.jsoup.nodes.Element;

import static io.pageobject.generator.Expressions.getFirstExpression;
import static io.pageobject.generator.attribute.Attributes.getNormalizedAttributeValue;

public class NgBindTemplateLocatorSource extends AbstractLocatorSource {

    @Override
    public String extractLocatorValue(Element element, GeneratorContext context) {
        return getFirstExpression(getNormalizedAttributeValue(element,
                                                              HtmlAttributes.NG_BIND_TEMPLATE.attributeName()));
    }

    public String locatorTemplate() {
        return "by.exactBinding('%s')";
    }

}

File: lib/locator/protractor/ng-bind-template-locator-source.js
"use strict";

const AbstractLocatorSource = require('../abstract-locator-source');
const _ = require('lodash');
const getNormalizedAttributeValue = require('../../attribute/attributes').getNormalizedAttributeValue;
const getFirstExpression = require('../../expressions').getFirstExpression;
const HtmlAttribute = require('../../attribute/html-attribute');

const template = _.template("by.exactBinding('<%= value %>')");

class NgBindTemplateLocatorSource extends AbstractLocatorSource {

  extractLocatorValue(context) {
    return getFirstExpression(getNormalizedAttributeValue(context.element,
                                                          HtmlAttribute.NG_BIND_TEMPLATE.attributeName));
  }

  locatorTemplate() {
    return template;
  }

}

module.exports = NgBindTemplateLocatorSource;